framework/CodeInterfaces/PHISICS/phisicsdata.py

```diff
@@ -0,0 +1,961 @@
+    """
+      Parses the absolute cross section output generated by PHISICS.
+      @ In, xsOutput, string, filename of the XML file containing the absolute perturbed XS
+      @ Out, labelList, list of the XS labels
+      @ Out, valueList, list of the XS
+    """
+    labelList = []
+    valueList = []
+    tree = ET.parse(os.path.join(self.workingDir,xsOutput))
+    root = tree.getroot()
+    for materialXML in root.getiterator('library'):
+      for isotopeXML in materialXML.getiterator('isotope'):
+        reactionList = [j.tag for j in isotopeXML]
+        for k in xrange (0, len(reactionList)):
+          for groupXML in isotopeXML.getiterator(reactionList[k]):
+            individualGroup = [x.strip() for x in groupXML.attrib.get('g').split(',')]
+            individualGroupValues = [y.strip() for y in groupXML.text.split(',')]
+            for position in xrange(0,len(individualGroup)):
+              labelList.append(materialXML.attrib.get('lib_name')+'|'+isotopeXML.attrib.get('id')+'|'+reactionList[k]+'|'+individualGroup[position])
+              valueList.append(individualGroupValues[position])
+    return labelList, valueList
+    
+  def removeSpaces(self,line):
+    """
+      Removes the spaces.
+      @ In, line, string
+      @ Out, line, string, same line without the blank spaces
+    """
+    line = re.sub(r' ',r'',line)
+    return line
+    
+  def getNumberOfGroups(self):
+    """
+      Gives the number of energy groups used in the PHISICS simulations.
+      @ In, None 
+      @ Out, None 
+    """
+    self.Ngroups = 0
+    with open(os.path.join(self.workingDir,self.instantOutputFileMPI[0]), 'r') as outfile:
+      for line in outfile:
+        if re.search(r'Number of elements of group', line):
+          self.Ngroups = line.split()[-3]  # pick the last occurence (last group), which is also the number of groups
+        if re.search(r'Number of element  types', line):
+          break
+  
+  def getNumberOfRegions(self):
+    """
+      Gives the number of spatial regions used in the PHISICS simulations.
+      @ Out, None
+    """
+    flagStart = 0
+    count = 0
+    subdomainList = []
+    with open(os.path.join(self.workingDir,self.instantOutputFileMPI[0]), 'r') as outfile:
+      for line in outfile:
+        if re.search(r'Subdomain volumes',line):
+          flagStart = 1
+        if re.search(r'Balance report for the primal solution', line):
+          flagStart = 2
+        if flagStart == 1:
+          stringIsNumber = self.isNumber(line.split())
+          if stringIsNumber is True:
+            count = count + 1
+            subdomainList.append(line.split()[0])
+        if flagStart == 2:
+          break
+    self.Nregions = count
+    return
+  
+  def getPertNumber(self):
+    """
+      Gets the current perturbation number.
+      @ In, None
+      @ Out, pertNumber, string, perturbation number in a string format
+    """
+    splitWorkDir = self.workingDir.split('/')
```

Comment:
```os.path.sep```

Reply:
I removed the method and defined the perturbation number in __initi__ at the begining